pkg/registry/apis/provisioning/jobs/queue.go

```diff
@@ -156,19 +156,21 @@ func (s *jobStore) drainPending() {
 			status, err = s.processByWorker(ctx, worker, *job)
 			if err != nil {
 				logger.Error("error processing job", "error", err)
-				status = &provisioning.JobStatus{
-					State:  provisioning.JobStateError,
-					Errors: []string{err.Error()},
+				if status == nil {
```

Comment:
[not for this PR] 
the way we return / handle statuses and errors is pretty messy and prone to errors. We keep tripping over ourselves because it doesn't follow the convention of not trusting the value if the function errors but trusting it if it's returned even if it's a pointer.